base/logger.go

```diff
@@ -34,3 +38,45 @@ func logCollationWorker(collateBuffer chan string, logger *log.Logger, bufferSiz
 		}
 	}
 }
+
+func runLogDeletion(logDirectory string, logLevel string, maxSize int) {
+
+	maxSize = maxSize * 1048576 //Convert MB input to bytes
+
+	logDirectory = logDirectory + "/"
+
+	logDiff := 0 - maxSize
+
+	files, err := ioutil.ReadDir(logDirectory)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	for _, f := range files {
+		var extension = filepath.Ext(logDirectory + f.Name())
+		if extension == ".gz" && strings.Contains(f.Name(), logLevel) {
+			logDiff += int(f.Size())
+		}
+	}
+
+	var modTime time.Time
+	var oldestFile string
+	var fileSize int
+	for logDiff > 0 {
+		files, _ = ioutil.ReadDir(logDirectory)
+		for _, f := range files {
+			var extension = filepath.Ext(logDirectory + f.Name())
+			if strings.Contains(f.Name(), logLevel) && extension == ".gz" {
+				if modTime.Before(f.ModTime()) || modTime.IsZero() {
```

Comment:
Don't we always set modTime to zero at the end of the loop? Wouldn't this be always true?

Reply:
We set it after the first loop. The outer loop loops until we are below the limit and the inner one finds the oldest file to delete.